src/Memory/BotState.ts

```diff
@@ -14,6 +14,32 @@ export interface ConversationSession {
     conversationId: string | null
 }
 
+export interface ConversationIdObject {
```

Comment:
This is all very Teams specific.  I wonder if we should call that out?

Reply:
Ok, added a comment.  

I thought if it was specific to teams the associated code should use something specific to teams like `if (channelId = "msteams")`.

I feel like these problems shouldn't need to be handled by the developer. That's whole point of having a single bot with multiple channels as the point of convergence.  Having to write custom code per channel makes you want to write another wrapper around them to converge

Reply:
Is this conversationIdObject? It looks like a partial activity to me.

Reply:
Yes, I was also confused.  I don't know why sometimes it's a string and other times it's an object. Lars thought it was a teams channel specific thing.

I see both these objects on the CONVERSATION_REFERENCE memory.  I think the CONVERSATION_ID is the one that is in question here, but I can't seem to find it reliably.

```
{"value":"\"19:3cfe1006d90344e79bb7cdad109222e5@thread.skype;messageid=1559938131876\"","eTag":"NaN"}
```
```
{"value":"{\"activityId\":\"1559944012861\",\"user\":{\"id\":\"29:1l6YV3UQRmmR5RP7OFY4kqVDGWJk_WDTpcjBtO0yEkI5Mg1oNFZxGKYBLTkRajNSzIPWhohWpXamjMqvViBD7Mg\",\"name\":\"Matt Mazzola\",\"aadObjectId\":\"ab2779a5-ff41-4c20-8efb-52d41070942e\"},\"bot\":{\"id\":\"28:5f4de912-a472-4450-a769-dc83add45603\",\"name\":\"teamsbug-bot02\"},\"conversation\":{\"isGroup\":true,\"conversationType\":\"channel\",\"tenantId\":\"72f988bf-86f1-41af-91ab-2d7cd011db47\",\"id\":\"19:3cfe1006d90344e79bb7cdad109222e5@thread.skype;messageid=1559943974486\"},\"channelId\":\"msteams\",\"serviceUrl\":\"https://smba.trafficmanager.net/amer/\"}","eTag":"NaN"}
```